contracts/__test__/contracts.test.ts

```diff
@@ -2,250 +2,219 @@ import { afterEach, beforeAll, beforeEach, describe, expect, test } from 'vitest
 
-const FEE_SINK_ADDR = 'A7NMWS3NT3IUDMLVO26ULGXGIIOUQ3ND2TXSER6EBGRZNOBOUIQXHIBGDE'
+describe('reti', () => {
+  const FEE_SINK_ADDR = 'A7NMWS3NT3IUDMLVO26ULGXGIIOUQ3ND2TXSER6EBGRZNOBOUIQXHIBGDE'
 
-const MaxPoolsPerNode = 3
-// Periodically set this to max amount allowed in protocol (200 atm) but when testing more frequently this should be lowered to something like 20 stakers
-// The ValidatorWFullPoolWRewards test is 'skip'ped for now - but should be periodically enabled for testing.
-const MaxStakersPerPool = 200
+  const MaxPoolsPerNode = 3
+  // Periodically set this to max amount allowed in protocol (200 atm) but when testing more frequently this should be lowered to something like 20 stakers
+  // The ValidatorWFullPoolWRewards test is 'skip'ped for now - but should be periodically enabled for testing.
+  const MaxStakersPerPool = 200n
 
-const fixture = algorandFixture({ testAccountFunding: AlgoAmount.Algos(10000) })
-const logs = algoKitLogCaptureFixture()
+  const fixture = algorandFixture({ testAccountFunding: AlgoAmount.Algos(10000) })
+  const logs = algoKitLogCaptureFixture()
 
-// algokit.Config.configure({ debug: true });
+  // algokit.Config.configure({ debug: true });
 
-const MaxAlgoPerPool = AlgoAmount.Algos(100_000).microAlgos
-let validatorMasterClient: ValidatorRegistryClient
-let poolClient: StakingPoolClient
+  const MaxAlgoPerPool = AlgoAmount.Algos(100_000).microAlgos
+  let validatorMasterClient: ValidatorRegistryClient
+  let validatorMasterAlgorandClient: AlgorandClient
+  let validatorFactory: ValidatorRegistryFactory
+  let stakingPoolFactory: StakingPoolFactory
+  let mbrs: MbrAmounts
 
-let validatorMbr: bigint
-let poolMbr: bigint
-let poolInitMbr: bigint
-let stakerMbr: bigint
-
-// =====
-// First construct the 'template' pool and then the master validator contract that everything will use
-beforeAll(async () => {
+  // =====
+  // First construct the 'template' pool and then the master validator contract that everything will use
+  beforeAll(async () => {
     await fixture.beforeEach()
+
     // testAccount here is the account that creates the Validator master contracts themselves - but basically one-time thing to be ignored
-    const { algod, testAccount } = fixture.context
+    const { algorand, testAccount } = fixture.context
+    validatorMasterAlgorandClient = algorand
 
     // Generate staking pool template instance that the validator registry will reference
-    poolClient = new StakingPoolClient(
-        {
-            sender: testAccount,
-            resolveBy: 'id',
-            id: 0,
-        },
-        algod,
-    )
-    const { approvalCompiled } = await poolClient.appClient.compile({
-        deployTimeParams: {
-            nfdRegistryAppId: 0,
-        },
+    stakingPoolFactory = validatorMasterAlgorandClient.client.getTypedAppFactory(StakingPoolFactory)
+    const { approvalProgram: stakingApprovalProgram } = await stakingPoolFactory.appFactory.compile({
```

Comment:
The *compiled* staking pool program has to be loaded into the validator contract.
Isn't this the 'teal' ? This can never work.

Reply:
FYI I changed this based on your feedback. I kept `approvalProgram` and `clearStateProgram` as the compiled code since adding "compiled" to a method call called compile is superfluous, but it's a proper type with jsdoc comments and the compilation result objects are named different to distinguish clearly (also with jsdoc comments).